src/time-range-picker/time-range-picker.tsx

```diff
@@ -150,64 +147,57 @@ export const TimeRangePicker = ({
                 $error={error}
                 $readOnly={readOnly}
             >
-                <InputSelectorStartTimeElement
+                <SelectorInput
                     onFocus={handleStartTimeFocus}
-                    $focused={showStartTimeSelector}
                     readOnly
-                    placeholder={"From"}
-                    value={TimeRangePickerHelper.formatValue(
-                        startTimeVal,
-                        format
-                    )}
+                    placeholder="From"
+                    value={TimeHelper.formatValue(startTimeVal, format)}
                     disabled={disabled}
-                    $error={error}
                     data-testid={
                         otherProps["data-testid"]
                             ? `${otherProps["data-testid"]}-timepicker-selector`
                             : "timepicker-selector"
                     }
                 />
-                {showStartTimeSelector && <BottomHighlightStartTime />}
-
-                <TimeRangePickerDropdown
+                <TimepickerDropdown
                     id={id}
                     show={showStartTimeSelector}
                     value={startTimeVal}
                     format={format}
                     onCancel={handleSelectionDropdownCancel}
                     onChange={handleStartTime}
                 />
-                <ArrowRangeContainer>
-                    <ArrowRight />
-                </ArrowRangeContainer>
-
-                {showEndTimeSelector && <BottomHighlightEndTime />}
-
-                <InputSelectorEndTimeElement
+                <ArrowRight />
+                <SelectorInput
                     onFocus={handleEndTimeFocus}
-                    $focused={showEndTimeSelector}
                     readOnly
-                    placeholder={"To"}
-                    value={TimeRangePickerHelper.formatValue(
-                        endTimeVal,
-                        format
-                    )}
+                    placeholder="To"
+                    value={TimeHelper.formatValue(endTimeVal, format)}
                     disabled={disabled}
-                    $error={error}
                     data-testid={
                         otherProps["data-testid"]
                             ? `${otherProps["data-testid"]}-timepicker-selector`
                             : "timepicker-selector"
                     }
                 />
-                <TimeRangePickerDropdown
+                <TimepickerDropdown
                     id={id}
                     show={showEndTimeSelector}
                     value={endTimeVal}
                     format={format}
                     onCancel={handleSelectionDropdownCancel}
```

Comment:
We're missing a `onSelectionCancel` callback on the component level

Reply:
what's the use of this callback? to detect the closing of the floating dropdown?

it doesn't seem useful for the ranged field since `onChange` only triggers on confirmation.

Reply:
It was to trigger the error message iirc. else can trigger an `onBlur` instead. If so, to standardise with the single input one.

Reply:
was planning to put `onSelectionCancel` back, but found out the cancel event also causes a blur event. instead of managing focus, i just standardised both to call `onBlur`